File: src/main/java/com/example/usercrud/business/entity/Role.java
package com.example.usercrud.business.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.security.core.authority.SimpleGrantedAuthority;

import java.util.Set;
import java.util.stream.Collectors;

@AllArgsConstructor
@Getter
public enum Role {
    USER(Set.of(Permission.CREATE_USER)),
    ADMIN(Set.of(Permission.CREATE_USER, Permission.EDIT_USER, Permission.GET_USER, Permission.DELETE_USER));

    private final Set<Permission> permissions;

    public Set<SimpleGrantedAuthority> getAuthorities() {
        return getPermissions().stream()
            .map(permission -> new SimpleGrantedAuthority(permission.getPermission()))
            .collect(Collectors.toSet());
    }
}

File: src/main/kotlin/cloud/folium/usercrud/business/entity/Role.kt
package cloud.folium.usercrud.business.entity

import org.springframework.security.core.authority.SimpleGrantedAuthority
import cloud.folium.usercrud.business.entity.UserRequestDto
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import org.springframework.web.multipart.MultipartFile
import java.util.function.Function
import java.util.stream.Collectors

enum class Role(val permissions: Set<Permission>) {
    USER(setOf(Permission.CREATE_USER)),
    ADMIN(setOf(Permission.CREATE_USER, Permission.EDIT_USER, Permission.GET_USER, Permission.DELETE_USER));

    fun getAuthorities(): Set<SimpleGrantedAuthority> {
        return permissions.stream().map{ permission -> SimpleGrantedAuthority(permission.permission)}
        .collect(Collectors.toSet())
    }
}